src/extension.ts

```diff
@@ -80,17 +82,55 @@ let decorationType: TextEditorDecorationType = window.createTextEditorDecoration
 
 const config = workspace.getConfiguration("metals");
 
+interface Activation {
+  shouldCreateNewProject: boolean;
+  status: Disposable;
+}
+
 export async function activate(context: ExtensionContext) {
   detectLaunchConfigurationChanges();
   checkServerVersion();
+  let activation: Activation = {
+    shouldCreateNewProject: false,
+    status: window.setStatusBarMessage("$(gear~spin) Activating Metals"),
```

Comment:
Added this to show the user that something is happening after clicking the `New Scala Project` button.

Reply:
Did you look into using `window.withProgress(..)` instead? (I'm not sure it makes sense, just throwing out ideas)

Reply:
I looked into it, but it seems the whole task needs to be included inside the `withProgress`. Haven't found a way yet to show/hide `withProgress` But that would be much better.

Reply:
I looked into it, but it seems the whole task needs to be included inside the `withProgress`. Haven't found a way yet to show/hide `withProgress` But that would be much better.

Reply:
Ok, actually coming back to it, changed to withProgress and added a few returns so `then` would actually work with a promise not void.